lib/widgets/message_list.dart

```diff
@@ -928,11 +900,12 @@ class DateSeparator extends StatelessWidget {
     const textBottomPadding = 2.0;
 
     final messageListTheme = MessageListTheme.of(context);
+    final designVariables = DesignVariables.of(context);
 
-    final line = BorderSide(width: 0, color: messageListTheme.dateSeparator);
+    final line = BorderSide(width: 0, color: designVariables.foreground);
```

Comment:
theme: Replace `dateSeparator` with `foreground`
    
    The variable giving color to divider lines in the Message List UI
    i.e. `dateSeparator` has been replaced with the `foreground` variable
    of `DesignVariables`, according to the Figma design.
    There isn't any change in the values of color from the previous one i.e.
    `dateSeparator`.

The fact there's no change in the visible behavior means this commit can be helpfully marked as NFC. See this description:
https://github.com/zulip/zulip-mobile/blob/main/docs/glossary.md#nfc

Reply:
Yeah, the main test for whether something is NFC is: is there any circumstance where a user could see different behavior because of the change?

If there is, then even if it's a subtle effect that only applies in an obscure case, that's a functional change and isn't NFC. But if there isn't, then that's NFC.

(Unless the prefix already says something else that means it's NFC in that sense, for example "test" or "lint". It'd be redundant to label all of those as NFC, so in that case we mark as NFC only if it has no effect on the behavior of the tests or lints etc., in a similar spirit though with different details.)